core/src/main/scala/org/apache/spark/rpc/netty/Dispatcher.scala

```diff
@@ -194,12 +194,32 @@ private[netty] class Dispatcher(nettyEnv: NettyRpcEnv, numUsableCores: Int) exte
     endpoints.containsKey(name)
   }
 
-  /** Thread pool used for dispatching messages. */
-  private val threadpool: ThreadPoolExecutor = {
+  def getNumOfThreads(conf: SparkConf): Int = {
     val availableCores =
       if (numUsableCores > 0) numUsableCores else Runtime.getRuntime.availableProcessors()
-    val numThreads = nettyEnv.conf.get(RPC_NETTY_DISPATCHER_NUM_THREADS)
+    // module configuration
+    val modNumThreads = nettyEnv.conf.get(RPC_NETTY_DISPATCHER_NUM_THREADS)
       .getOrElse(math.max(2, availableCores))
+    // try to get specific threads configurations of driver and executor
+    // override module configurations if specified
+    val executorId = conf.get("spark.executor.id", "")
```

Comment:
You should use `conf.get(EXECUTOR_ID)`. But it seems there's a problem here.

If you look at the initialization of `SparkEnv`, "spark.executor.id" is set in the configuration after the RpcEnv (and thus this class) is initialized.

Reply:
It seems you are true. let me double-check. thanks.

Reply:
for driver, we can get spark.executor.id with value of "driver", but not for executor. let me fix it.

Reply:
@vanzin , I set executorId to spark.executor.id explicitly in two executors' backends, CoarseGrainedExecutorBackend and MesosExecutorBackend. For yarn, it uses CoarseGrainedExecutorBackend. For driver, we don't need to set it since it's already set.